### ISSUE TITLE
#Build fail on Ubuntu 16.04

### ISSUE THREAD

**Issue by eduardoxy** [deeplearning4j/libnd4j#902]
Hello!

I'm trying to install the libnd4j in my machine.
I get some errors that I've solved.

But there is an error avoiding the successful build.

The main line of the error is above:
/usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.

I set the CMakeLists.txt with 
set(CMAKE_CXX_FLAGS "-std=c++11 -O3")
and a rm -rf on the built directory was made.
But it has no effect. The same error persists.

I sent the log copied by the terminal to help you to find some solution.
Thanks

[error.txt](https://github.com/deeplearning4j/libnd4j/files/2011636/error.txt)

**Comment by raver119** [comment]
Let's start from the beginning please.

What's your compiler? 

We have at least 2 team members using Ubuntu 16.04, and i know that there's no problems with problem compiler.

**Comment by eduardoxy** [comment]
My fault...
Sorry

I was using the gcc 5.2.
I understood that the compiler must be 4.9 or higher instead of only the 4.9.
I will remove and reinstall the correct gcc version.
When the process ends I will tell you the results.

Thank you for now

**Comment by raver119** [comment]
4.9 should work fine too, we use it for Android builds. But in reality for casual x86_64 system i'd recommend using recent compiler :)

**Comment by eduardoxy** [comment]
Well,

After put a new gcc and run all related commands I've tried again.
Unfortunately I get a similar error.
First I ran ./buildnativeoperations.sh
It completes without any error.
But when I run ./buildnativeoperations.sh -c cuda -cc 20
(I have a GeForce 610M)
I get this result:

/usr/include/c++/4.9/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \

**Comment by eduardoxy** [comment]
My dependences have the following versions:
#####################
eduardoxy@eduardoxy-asus:~/Documents/workspace/libnd4j$ gcc --version
gcc (Ubuntu 4.9.4-2ubuntu1~16.04) 4.9.4
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

eduardoxy@eduardoxy-asus:~/Documents/workspace/libnd4j$ nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2015 NVIDIA Corporation
Built on Tue_Aug_11_14:27:32_CDT_2015
Cuda compilation tools, release 7.5, V7.5.17
eduardoxy@eduardoxy-asus:~/Documents/workspace/libnd4j$ cmake --version
cmake version 3.9.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).

**Comment by eduardoxy** [comment]
Hello friend,

Finally I could compile the libndj4 for the cuda.
The problem was solved.
I was installing the Cuda Toolkit using the apt-get command and the apt-get command installed the cuda 7.5 version.

I went to the nvidia website and download the cuda 9.0 version. Once I installed it manually the libndj4 code compiled without errors.

If someone faces the same errors, I am leaving my setup below:
Ubuntu 16.04 x64
GeForce GT610M
GCC 4.9.4
CMake 3.9
Cuda 9.0

Thank you for your support.
I will close the issue because it is solved.
Best regards
